packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,255 @@
+// @ts-ignore
+import * as jwt from 'jsonwebtoken';
+import { utils, providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+import { StorageSettings } from './models';
+import { CredentialResolver } from '.';
+import { IVerifiableCredential, VerificationResult } from './models';
+
+export class IssuanceVerification {
+  private _provider: providers.Provider;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _roleDefResolver: RoleDefinitionResolverV2;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param roleDefResolverAddr
+   * @param registrySetting
+   * @param storageSettings
+   */
+  constructor({
+    provider,
+    roleDefResolverAddr,
+    registrySetting,
+    storageSettings,
+  }: {
+    provider: providers.Provider;
+    roleDefResolverAddr: string;
+    registrySetting: RegistrySettings;
+    storageSettings: StorageSettings;
+  }) {
+    this._provider = provider;
+    this._registrySetting = registrySetting;
+    this._roleDefResolver = RoleDefinitionResolverV2__factory.connect(
+      roleDefResolverAddr,
+      this._provider
+    );
+    this._resolver = new Resolver(this._provider, this._registrySetting);
+    this._credentialResolver = new CredentialResolver(storageSettings);
+  }
+
+  /**
+   * Verifies chain of trust for a holder's credential
+   * @param credential
+   * @param verifyCredentialProofCallback
+   * @param subjectDID
+   * @param role
+   * @returns
+   */
+  async verifyChainOfTrust(
+    credential?: IVerifiableCredential,
+    verifyCredentialProofCallback?: (
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>,
+    subjectDID?: string,
+    role?: string
+  ) {
+    if (credential && verifyCredentialProofCallback) {
+      this.verifyCredentialChainOfTrust(
+        credential,
+        verifyCredentialProofCallback
+      );
+    } else if (subjectDID && role) {
+      this.verifySubjectChainOfTrust(subjectDID, role);
+    } else {
+      return 'Insufficient information';
+    }
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's DID and role
+   * @param {string} subjectDID
+   * @param {string} role
+   */
+  async verifySubjectChainOfTrust(subjectDID: string, role: string) {
```

Comment:
ok, do we need to support `LegacyEWCredential` within this verification code. I see this `verification` functionality been consumed in our VC presentation and other verifiable credential operations as per latest specs, I think it's better to enforce the latest changes and restrict it to verifiable credential only. Since the legacy credential already has verification mechanism and we would eventually discard them. What do you think?

Reply:
Yeah, it would be a good approach and then it could be deprecated whenever needed.